util/post-receive.git.hook.example

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+
+# You may need to customize these for your environment if the git user doesn't
+# have the `go` cmd in their $PATH or if the git user has a non-standard GOPATH
+GOCMD=go
+GOPATH=$GOPATH
+
+while read oldrev newrev ref
+do
+  if [[ $ref =~ .*/master$ ]];
+  then
+    echo "Master ref received. Deploying master branch update."
+    pushd $GOPATH/src/github.com/letsencrypt/ct-woodpecker
+      git --work-tree=./ --git-dir=/usr/local/src/ct-woodpecker.git checkout -f
+      unset GIT_DIR
```

Comment:
Why would GIT_DIR be set?

Reply:
Per the [githook docs](https://git-scm.com/docs/githooks):
> Before Git invokes a hook, it changes its working directory to either $GIT_DIR in a bare repository or the root of the working tree in a non-bare repository. An exception are hooks triggered during a push (pre-receive, update, post-receive, post-update, push-to-checkout) which are always executed in $GIT_DIR.

In this specific case `GIT_DIR` is set to the bare repo (`/usr/local/src/ct-woodpecker.git`) when the post-receive hook is run. The `git checkout` we run in the hook works around that by expressly overriding it with `--git-dir` to make sure the checkout works correctly. We can't tell the `go get` in our Makefile to use an explicit `--git-dir` and so without unsetting `GIT_DIR` calling `make install` from the hook fails because the `go get` ends up bailing because the bare repo doesn't have a `remote` origin.

Reply:
I'll try removing the `--git-dir` in the `git checkout` tomorrow. I'm out of time to do a test deploy with the diff tonight.